src/dbSta/src/dbNetwork.cc

```diff
@@ -604,6 +622,18 @@ dbNetwork::netIterator(const Instance* instance) const
 
 ////////////////////////////////////////////////////////////////
 
+ObjectId dbNetwork::id(const Pin* pin) const
+{
+  dbITerm* iterm;
+  dbBTerm* bterm;
+  staToDb(pin, iterm, bterm);
+
+  if (iterm) {
+    return iterm->getId() + iterm->getBlock()->getBTerms().size();
```

Comment:
why add the bterm size?

Reply:
A sta pin maps to either a bterm or an iterm.  To give a unique id I can't just use the odb id for each as they would overlap.  I offset the iterms ids to make them unique.  I could have done the reverse and offset the bterms by the #iterms.